### ISSUE TITLE
Support ignoring updates/tracking of objects with specific annotation

### ISSUE THREAD

**Issue by clearnicki** [argoproj/argo-cd#21169]
# Summary

Have a way to explicitly ignore/orphan certain objects.

# Motivation

If an object is being updated and tracked by argo, we would like to be able to have a way to easily orphan that object without needing to remove it from git or remove tracking labels.  We have had instances where multiple applications keep trying to update an object even when the tracking label is removed, since multiple apps depend on this particular object. 

# Proposal

We would like to be able to add an annotation to the object to force argo to ignore the object. This would take precedence over any other tracking label or argo annotation.

**Comment by andrii-korotkov-verkada** [comment]
Why not remove it from Git? That seems like a better way, given ArgoCD's focus on declarative setup.

**Comment by agaudreault** [comment]
I agree with @andrii-korotkov-verkada. If Argo is touching that object, it means it is in Git. Then it would go directly against what Argo is meant for: maintaining a source of truth.

If multiple applications depends on the same object, I would suggest you to update one (or all) of your applications to exclude that object, and deploy is separately. This can be done with Kustomize with a delete patch and can be combined with Helm too by rendering the charts with Kustomize.
